src/Concordium/Client/Output.hs

```diff
@@ -699,6 +728,26 @@ showEvent verbose = \case
     showDelegationTarget Types.DelegatePassive = "Passive delegation"
     showDelegationTarget (Types.DelegateToBaker bid) = "Baker ID " ++ show bid
 
+    showLoggedEvents :: [Wasm.ContractEvent] -> String
+    showLoggedEvents [] = "No contract events were emitted "
+    showLoggedEvents evs = [i|#{length evs} contract events were emitted|]
+      <> if isNothing stM
+         then [i| but no event schema was provided nor found in the contract module. |]
+         else [i|, of which #{length $ filter isJust $ map snd loggedEvents} were succesfully parsed using schema. |]
+      <> [i|Got:\n|]
+      <> intercalate "\n" [ if isJust json
+                            then [i|Event(parsed):\n#{indentBy 4 $ fromMaybe' json}|]
+                            else [i|Event(raw): #{hex}\n|]
+                            | (hex, json) <- loggedEvents ]
+        where
+          fromMaybe' :: Maybe AE.Value -> String
+          fromMaybe' = \case Nothing -> "Unable to decode."; Just j -> showSortedPrettyJSON j
+          toHex :: Wasm.ContractEvent  -> String
+          toHex (Wasm.ContractEvent bs) = show . BSB.toLazyByteString . BSB.byteStringHex $ SE.runPut $ SE.putShortByteString bs
+          toJSON' :: Wasm.ContractEvent -> Maybe AE.Value
+          toJSON' (Wasm.ContractEvent bs) =
+            stM >>= \st' -> (preview _Right . PA.decodeParameter st' . SE.runPut . SE.putShortByteString) bs
+          loggedEvents = map (bimap toHex toJSON') (zip evs evs)
```

Comment:
This is a very complex way of writing it and I do not see why it needs to be. Why can't we just do

```haskell
eventToString e = case toJSON' e of
    Nothing -> [i|Event(raw): #{toHex e}\n|]
    Just json -> [i|Event(parsed):\n#{indentBy 4 $ json}|]
```

And then instead of the `intercalate` just to `unlines . map eventToString $ evs` or similar.

It'll also remove the strange `fromMaybe'`.